src/pgrx_tests/common.rs

```diff
@@ -0,0 +1,307 @@
+                let actual = actual.unwrap();
+
+                let expected_key: Option<i32> = expected.get_by_name("key").unwrap();
+                let actual_key: Option<i32> = actual.get_by_name("key").unwrap();
+
+                assert_eq!(expected_key, actual_key);
+
+                let expected_val: Option<String> = expected.get_by_name("val").unwrap();
+                let actual_val: Option<String> = actual.get_by_name("val").unwrap();
+
+                assert_eq!(expected_val, actual_val);
+            }
+        }
+    }
+}
+
+pub(crate) struct TestResult<T> {
+    pub(crate) expected: Vec<(Option<T>,)>,
+    pub(crate) result: Vec<(Option<T>,)>,
+}
+
+pub(crate) fn test_common<T: IntoDatum + FromDatum>(test_table: TestTable<T>) -> TestResult<T> {
+    test_table.copy_to_parquet();
+    test_table.copy_from_parquet();
+
+    let expected = test_table.select_all("test_expected");
+    let result = test_table.select_all("test_result");
+
+    TestResult { expected, result }
+}
+
+pub(crate) fn test_assert<T>(expected_result: Vec<(Option<T>,)>, result: Vec<(Option<T>,)>)
+where
+    T: Debug + PartialEq,
+{
+    for (expected, actual) in expected_result.into_iter().zip(result.into_iter()) {
+        assert_eq!(expected, actual);
+    }
+}
+
+pub(crate) fn test_assert_float(expected_result: Vec<Option<f32>>, result: Vec<Option<f32>>) {
+    for (expected, actual) in expected_result.into_iter().zip(result.into_iter()) {
+        if expected.is_none() {
+            assert!(actual.is_none());
+        }
+
+        if expected.is_some() {
+            assert!(actual.is_some());
+
+            let expected = expected.unwrap();
+            let actual = actual.unwrap();
+
+            if expected.is_nan() {
+                assert!(actual.is_nan());
+            } else if expected.is_infinite() {
+                assert!(actual.is_infinite());
+                assert!(expected.is_sign_positive() == actual.is_sign_positive());
+            } else {
+                assert_eq!(expected, actual);
+            }
+        }
+    }
+}
+
+pub(crate) fn test_assert_double(expected_result: Vec<Option<f64>>, result: Vec<Option<f64>>) {
+    for (expected, actual) in expected_result.into_iter().zip(result.into_iter()) {
+        if expected.is_none() {
+            assert!(actual.is_none());
+        }
+
+        if expected.is_some() {
+            assert!(actual.is_some());
+
+            let expected = expected.unwrap();
+            let actual = actual.unwrap();
+
+            if expected.is_nan() {
+                assert!(actual.is_nan());
+            } else if expected.is_infinite() {
+                assert!(actual.is_infinite());
+                assert!(expected.is_sign_positive() == actual.is_sign_positive());
+            } else {
+                assert_eq!(expected, actual);
+            }
+        }
+    }
+}
+
+pub(crate) fn test_helper<T: IntoDatum + FromDatum + Debug + PartialEq>(test_table: TestTable<T>) {
```

Comment:
could be nice to have a more descriptive name here now that things are getting more dispersed, e.g. test_export_import_equality test_round_trip_equality